src/beacon/app/device_performance_endpoint.py

```
@@ -13,202 +13,389 @@
     expected_readings_per_day = 24  # Assuming hourly readings
     expected_total_readings = days_in_range * expected_readings_per_day
     
-    # Group readings by day to analyze data completeness
+    # Group readings by day for comprehensive analysis
+    daily_readings = group_readings_by_day(readings_data, start_date, end_date)
+    
+    # Calculate daily metrics using ALL sensor data
+    daily_data = calculate_comprehensive_daily_metrics(daily_readings, expected_readings_per_day)
+    
+    # Calculate sensor-specific performance metrics
+    sensor_metrics = calculate_sensor_performance_metrics(readings_data)
```

Ruff (0.12.2)

Undefined name `calculate_sensor_performance_metrics`

(F821)